src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5082,8 +5082,8 @@ MacroAssembler::KlassDecodeMode MacroAssembler::klass_decode_mode() {
 
   if (operand_valid_for_logical_immediate(
         /*is32*/false, (uint64_t)CompressedKlassPointers::base())) {
```

Comment:
Reviewer hint:

The before and after code should be exactly identical. What we do here is we calculate the max. value a left-shifted nKlass can have, and compute a mask over it.